pkg/util/ciutil/gitlab.go

```diff
@@ -27,13 +27,13 @@ type gitlabCI struct {
 // See https://docs.gitlab.com/ee/ci/variables/.
 func (gl gitlabCI) DetectVars() Vars {
 	v := Vars{Name: gl.Name}
-	v.BuildID = os.Getenv("CI_JOB_ID")
+	v.BuildID = os.Getenv("CI_PIPELINE_IID")
 	v.BuildType = os.Getenv("CI_PIPELINE_SOURCE")
 	v.BuildURL = os.Getenv("CI_JOB_URL")
 	v.SHA = os.Getenv("CI_COMMIT_SHA")
 	v.BranchName = os.Getenv("CI_COMMIT_REF_NAME")
 	v.CommitMessage = os.Getenv("CI_COMMIT_MESSAGE")
-	v.PRNumber = os.Getenv("CI_MERGE_REQUEST_ID")
+	v.PRNumber = os.Getenv("CI_MERGE_REQUEST_IID")
```

Comment:
The previous env var was outright incorrect since it was the GitLab internal ID, which cannot be used to navigate to the specific GitLab merge request. `CI_MERGE_REQUEST_IID` is the merge request instance ID, again specific to the current project and can be used to identify a merge request in a project.